dcos/subcommand.py

```diff
@@ -435,3 +465,77 @@ def package_json(self):
         package_json_path = os.path.join(self._dir(), 'package.json')
         with util.open_file(package_json_path) as package_json_file:
             return util.load_json(package_json_file)
+
+
+class SubcommandMain():
+    """Representes a subcommand running in the main thread"""
+
+    def __init__(self, command):
+        """
+        :param commad: function to run in thread
+        :type command: func
```

Comment:
This looks like it is a `str`. It is only in `run_and_capture` that we look for the func.